bp_common/test/src/bootrom/bootrom.S

```diff
@@ -35,13 +50,43 @@ _start:
     li x30, 0
     li x31, 0
 
-    # Enable FPU
-    li x31, 0x2000
-    csrs mstatus, x31
+    // Copy CCE microcode
+    la t0, _ucode
+    li t1, CFG_CCE_UCODE_BASE_ADDR
+    li t2, CCE_INSTR_END
+load_ucode:
+    ld t3, 0(t0)
+    beq t3, t2, load_config
+    sd t3, 0(t1)
+    addi t0, t0, 8
+    addi t1, t1, 1
+    j load_ucode
+        
+    // Switch I$, D$ and CCE modes
+load_config:
+    li t0, 1
+    li t1, CFG_DCACHE_MODE_ADDR
+    sd t0, 0(t1)
+    
+    li t1, CFG_ICACHE_MODE_ADDR
+    sd t0, 0(t1)
+
+    li t1, CFG_CCE_MODE_ADDR
+    sd t0, 0(t1)
     
```

Comment:
Should have a fence here, I think

Reply:
Well, these are stores to uncached addresses. I don't think a fence will have any effect here. Is that right?

Reply:
That is not correct.  A fence ensures that the store completes before moving to the next instruction